### PR TITLE
Assembler: Update zoom control to be a button instead of a slider 

### PR DESCRIPTION
**Pull request by taipeicoder**
Related to #85123

## Proposed Changes

This PR updates the zoom control from a slider to a button. Clicking the button will zoom out the device viewport by 50%. 

| Before | After |
| --- | --- | 
| Screenshot 2023-12-27 at 12 04 23 PM | Screenshot 2023-12-27 at 12 04 51 PM |

Final version:

https://github.com/Automattic/wp-calypso/assets/797888/9f4d96b8-f8b7-4ad8-9092-5573a0fc561a

## Testing Instructions

* Head to the Assembler.
* Select any pattern.
* Ensure that the zoom toggle is visible.
* Ensure that the zoom toggle works as described above.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by arthur791004 — APPROVED** [review]
Tested and looks good to me!

Feedback:
* It would be better to add the tooltip to explain what's this, e.g.: Zoom out to 50%.
* I think the global view is just to zoom out to 50% based on the current viewport. That means users can zoom out the preview on the mobile view as well. However, the current global view is always based on the desktop view, so people cannot have a global view on mobile.

**Comment by taipeicoder** [comment]
> It would be better to add the tooltip to explain what's this, e.g.: Zoom out to 50%.
👍 

> I think the global view is just to zoom out to 50% based on the current viewport. That means users can zoom out the preview on the mobile view as well. However, the current global view is always based on the desktop view, so people cannot have a global view on mobile.

That's a good point. To confirm, you mean 50% of the selected device, right?

**Comment by arthur791004** [comment]
> That's a good point. To confirm, you mean 50% of the selected device, right?

Yes.

**Comment by taipeicoder** [comment]
> > That's a good point. To confirm, you mean 50% of the selected device, right?
> 
> Yes.

I've updated the logic as suggested. LMK if this is what you had in mind!

File: packages/components/src/device-switcher/fixed-viewport.tsx [hunk 1]
```diff
@@ -11,16 +12,28 @@ const deviceWidthById = {
 	phone: DEVICE_PHONE_WIDTH,
 } as Record< string, number >;
 
-export const useViewportScale = ( device: string, viewportWidth: number ) => {
-	const deviceWidth: number = deviceWidthById[ device ];
+const getDeviceWidth = ( device: string, previousDevice: string ) => {
+	if ( DEVICE_TYPES.GLOBAL_VIEW === device && deviceWidthById[ previousDevice ] ) {
+		return deviceWidthById[ previousDevice ] / 2;
```
Comment (arthur791004) [inline packages/components/src/device-switcher/fixed-viewport.tsx]: Why do we want to device it by 2 🤔

**Review by arthur791004 — COMMENTED** [review]
> I've updated the logic as suggested. LMK if this is what you had in mind!

Yeah, similar. But I think the global view is just a toggle between 50% and 100% zoom out, and we can keep focusing on the current device. If people click the global view again, we can reset the zoom out ratio to 100%. However, maybe it's out of scope and different than the original idea 😅

File: packages/components/src/device-switcher/fixed-viewport.tsx (hunk 1 above)
Comment (taipeicoder) [reply packages/components/src/device-switcher/fixed-viewport.tsx]: It's not needed, I forgot to remove it 🫠

**Comment by taipeicoder** [comment]
> Yeah, similar. But I think the global view is just a toggle between 50% and 100% zoom out, and we can keep focusing on the current device. If people click the global view again, we can reset the zoom out ratio to 100%. However, maybe it's out of scope and different than the original idea 😅

Let's keep the following the behavior until we have further feedback: 
Clicking the zoom button will now zoom out 50% the device the user has previously selected. In other words, if users selected the desktop viewport and click on zoom, we will zoom out 50% of the desktop viewport. The same applies for the tablet, and mobile viewports. See recording for a demo:

https://github.com/Automattic/wp-calypso/assets/797888/0e9b94eb-e696-4d63-993c-d4262561fc20

@arthur791004 proposes the zoom button to be a toggle button that can be activated together with any device button. This means that users will see both the desktop and the zoom button selected simultaneously. This is more accurate with the behavior of the zoom, IMO 🙂

Thoughts @autumnfjeld @Automattic/lego @matt-west? @lucasmendes-design is AFK so we can iterate when he gets back.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10513641

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @taipeicoder for including a screenshot in the description! This is really helpful for our translators.

**Comment by autumnfjeld** [comment]
Assuming I understand the conversation above, it seems fine to make both the device icon and the zoom icon appear as "selected".     

To me these are minor fixes that we can easily update next week.   So I'm fine with shipping as is or making the change suggested by @arthur791004 .

**Comment by taipeicoder** [comment]
Here's a recording of the updated behavior:

https://github.com/Automattic/wp-calypso/assets/797888/7ec4f793-f1fe-4cb8-82db-d04c9cdd427f

@arthur791004 I've re-requested a review since the implementation was significantly changed.

**Review by arthur791004 — APPROVED** [review]
Still looks good 👍

**Comment by matt-west** [comment]
@taipeicoder Here are some icons for the zoom button. We’ll start with the `zoom-out` icon, then switch that for `zoom-in` when the the preview is at 50% zoom. These should be displayed at 18px by 18px.

zoom-out
zoom-in

0JnQCbIpPLeXr9NOOLqdlD-fi-436_11467

**Comment by taipeicoder** [comment]
> @taipeicoder Here are some icons for the zoom button. We’ll start with the `zoom-out` icon, then switch that for `zoom-in` when the the preview is at 50% zoom. These should be displayed at 18px by 18px.
> 
> zoom-out zoom-in
> 
> [0JnQCbIpPLeXr9NOOLqdlD-fi-436_11467](https://href.li/?https://www.figma.com/file/0JnQCbIpPLeXr9NOOLqdlD/?node-id=436-11467)

Thank you @matt-west! I've updated the icons, so you can check them out via the calypso.live link in this [comment](https://github.com/Automattic/wp-calypso/pull/85787#issuecomment-1869925323). I noticed that the icon looks a bit low fidelity in my monitor:
Screenshot 2024-01-03 at 11 01 58 AM

Seems that there are some antialiasing issues with the circle and the plus sign. Is there any way we can improve that?

File: packages/components/src/device-switcher/icons.tsx [hunk 2]
```diff
@@ -20,3 +20,25 @@ export const phone = (
...
+export const zoomOut = (
+	<SVG width="18" height="18" viewBox="0 0 18 18" fill="none">
+		<Path
+			d="M7.5 13.5C10.8137 13.5 13.5 10.8137 13.5 7.5C13.5 4.18629 10.8137 1.5 7.5 1.5C4.18629 1.5 1.5 4.18629 1.5 7.5C1.5 10.8137 4.18629 13.5 7.5 13.5ZM7.5 15C9.30092 15 10.9536 14.3652 12.2465 13.3073L16.9393 18.0001L18 16.9395L13.3072 12.2466C14.3652 10.9537 15 9.30098 15 7.5C15 3.35786 11.6421 0 7.5 0C3.35786 0 0 3.35786 0 7.5C0 11.6421 3.35786 15 7.5 15ZM10.5 6.75H4.5V8.25H10.5V6.75Z"
+			fillRule="evenodd"
+			clipRule="evenodd"
+			fill="currentColor"
+		/>
+	</SVG>
+);
```
Comment (matt-west) [inline packages/components/src/device-switcher/icons.tsx]:
```suggestion
export const zoomIn = (
	<SVG width="24" height="24" viewBox="0 0 24 24">
		<Circle cx="11" cy="11" r="7.25" stroke="currentColor" stroke-width="1.5" />
		<Path d="M8 11H14M11 8V14" stroke="currentColor" stroke-width="1.5" />
		<Path d="M16 16L20 20" stroke="currentColor" stroke-width="1.5" />
	</SVG>
);

export const zoomOut = (
	<SVG width="24" height="24" viewBox="0 0 24 24">
		<Circle cx="11" cy="11" r="7.25" stroke="currentColor" stroke-width="1.5" />
		<Path d="M16 16L20 20" stroke="currentColor" stroke-width="1.5" />
		<Path d="M8 11H14" stroke="currentColor" stroke-width="1.5" />
	</SVG>
);
```

File: packages/components/src/device-switcher/icons.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { SVG, Rect } from '@wordpress/components';
+import { SVG, Path, Rect } from '@wordpress/components';
```
Comment (matt-west) [inline packages/components/src/device-switcher/icons.tsx]:
```suggestion
import { SVG, Path, Rect, Circle } from '@wordpress/components';
```

File: packages/components/src/device-switcher/toolbar.tsx [hunk 4]
```diff
@@ -38,6 +48,15 @@ const DeviceSwitcherToolbar = ( { device: currentDevice, onDeviceClick }: Toolba
 						/>
 					</Button>
 				) ) }
+				{ isZoomable && (
+					<Button
+						aria-label={ translate( 'Zoom out 50%' ) }
+						className={ classnames( { zoom: true, 'is-selected': isZoomActive } ) }
+						onClick={ () => onZoomClick() }
+					>
+						<Icon icon={ isZoomActive ? zoomIn : zoomOut } size={ 18 } />
```
Comment (matt-west) [inline packages/components/src/device-switcher/toolbar.tsx]:
```suggestion
						<Icon icon={ isZoomActive ? zoomIn : zoomOut } size={ 24 } />
```

**Review by matt-west — COMMENTED** [review]
Good catch @taipeicoder! I’ve added some suggested changes with the update icons. These should display better at lower resolutions.

I’ve also changed the icons to be 24 x 24, with some internal spacing so they’re consistent with the rest of the Dotcom icons.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10513641).

**Comment by taipeicoder** [comment]
> Good catch @taipeicoder! I’ve added some suggested changes with the update icons. These should display better at lower resolutions.
> 
> I’ve also changed the icons to be 24 x 24, with some internal spacing so they’re consistent with the rest of the Dotcom icons.

Looks much better, thanks @matt-west! I've attached a recording for reference:

https://github.com/Automattic/wp-calypso/assets/797888/3ed84637-d64c-416d-a840-b1c8dea30fee

**Pull request merged**